packages/nextalign/src/io/kseqpp.h

```diff
@@ -120,10 +116,9 @@ namespace klibpp {
       }
 
       while ((c = this->getc()) && c != '>') {
-        if (!is_char_allowed(c)) {
+        if (c == '\n') {
           continue;
         }
-        c = static_cast<char_type>(std::toupper(c));
         rec.seq += c;
         this->getuntil(KStream::SEP_LINE, rec.seq, nullptr, true);// read the rest of the line
       }
```

Comment:
The conditional was skipping the invalid characters only in the beginning, and `getuntil()` would copy all characters until the next newline. This is incorrect.
I removed this check entirely and moved sanitization in a separate step.